.github/workflows/build-and-test.yaml

```diff
@@ -2,14 +2,13 @@ name: build-and-test
 
 on:
   push:
-    branches:
-      - ${{ github.event.repository.default_branch }}
   schedule:
     - cron: 0 19 * * * # run at 4 AM JST
   workflow_dispatch:
 
 jobs:
   build-and-test:
+    if: ${{ github.ref_name == github.event.repository.default_branch }}
```

Comment:
For `workflow_dispatch`, we should add an additional condition to check if it's `push` event.

Reply:
Like this? :thinking: 

```suggestion
    if: ${{ github.event_name != 'pull_request' || github.ref_name == github.event.repository.default_branch }}
```

Reply:
@KeisukeShima Do you have any suggestions?

Reply:
How about https://github.com/autowarefoundation/autoware_common/pull/24/commits/8ca25f9037c92d5f3dbd8579a790f93094fe02ef ?

Reply:
Oh, it was `push`, not `pull_request`. I'm sorry.

It looks good to me, but could you link your test result here?

Reply:
Workflow dispatch result is here https://github.com/h-ohta/autoware_common/actions/runs/1805053193.

Reply:
Cool, thank you!

Reply:
How about using workflow_dispatch event?

https://github.community/t/trigger-job-only-when-workflow-dispatch-is-used/155112

```
    if: ${{ github.event_name == 'workflow_dispatch' || github.ref_name == github.event.repository.default_branch }}
```

Reply:
Thank you. I think it's fine as well. But since `push` has no problem, I'll keep it.